src/histogram.rs

```diff
@@ -41,9 +48,34 @@ fn check_bucket_label(label: &str) -> Result<()> {
     Ok(())
 }
 
-fn check_and_adjust_buckets(mut buckets: Vec<f64>) -> Result<Vec<f64>> {
+fn check_and_adjust_int_buckets(mut buckets: Vec<u64>) -> Result<Vec<u64>> {
```

Comment:
Maybe we can use generics to avoid writing code twice.

Reply:
From what i can tell this would require a bit of a re-architecture. I will take a closer look this may be a breaking change and see what i can do.